### DISCUSSION TITLE
Creating Rules to check Request Method

### DISCUSSION THREAD

**Discussion by yuuki85** [mockoon/mockoon#1511] [Help] [answered]
Is there any way to create rule for a route that will check if the request is the correct method? 
For example: (This example uses HTTP route)
route GET “user/:id”, it needs to be a GET method. If it is not, it should return a specific response with Response code 405.

Any help or tips would be greatly appreciated.

**Comment by 255kb** [comment, marked as answer]
When you create a GET route, for example, you have to use the GET method to call it, so it's already indirectly doing the check.

![image](https://github.com/user-attachments/assets/349c8623-98ca-47d1-99e7-f1bc2252ff46)

In the case you use the "All methods" menu entry, or a [CRUD route](https://mockoon.com/docs/latest/api-endpoints/crud-routes/), there is indeed no way to check the method with a rule and give a specific response. 
There is an [open issue](https://github.com/mockoon/mockoon/issues/1467), and I plan to add this to the next release (coming weeks).

**Reply by yuuki85** [reply]
>There is an https://github.com/mockoon/mockoon/issues/1467, and I plan to add this to the next release (coming weeks).

I see, I will be looking forward to it then.

Thank you.
